client/src/main.rs

```diff
@@ -174,9 +180,15 @@ fn main() {
         .add_cmd(
             Command::new("print-metadata")
                 .description("query node metadata and print it as json to stdout")
-                .runner(|_args: &str, matches: &ArgMatches<'_>| {
-                    let meta = get_chain_api(matches).get_metadata().unwrap();
-                    println!("Metadata:\n {}", Metadata::pretty_format(&meta).unwrap());
+                .runner(|_args: &str, _matches: &ArgMatches<'_>| {
+                    // Most likely upstream bug, because a trait bound is specified, but the struct does not derive it.
+                    // let meta = get_chain_api(matches).metadata();
+                    // let buf = Vec::new();
+                    // let formatter = serde_json::ser::PrettyFormatter::with_indent(b" ");
+                    // let mut ser = serde_json::Serializer::with_formatter(buf, formatter);
+                    // meta.serialize(&mut ser).unwrap();
+                    // let res = String::from_utf8(ser.into_inner());
+                    // println!("Metadata:\n {}", res.unwrap());
```

Comment:
Could also be a missing feature flag.

Reply:
@haerdib Could you please have a look at this? I think this might be an issue, because `Metadata` is defined like this:
``` rust
#[derive(Clone, Debug, Encode, Decode)]
pub struct Metadata {
```
and the `pretty_format` function is:
```rust
#[cfg(feature = "std")]
pub fn pretty_format<Metadata: Serialize>(metadata: &Metadata) -> Option<String> {
```
But the `Serialize` trait is not derived for it, and because of the orphan rule it is not possible for lib users to implement the trait either.

Reply:
Ah, my apologies, that's a real hassle of a naming issue and not very nicely implemented `pretty_format` function. It's only fixed in the later versions (https://github.com/scs/substrate-api-client/pull/532).

The issue here is, that the `Metadata` struct itself does not support Serialize, because that would simply over blow everything as it does contain all metadata plus the maps of events, errors and everything. To serialize the metadata, use the underlying substrate `runtime_metadata`. 

E.g. that should work: 

```rust
 let meta = get_chain_api(matches).metadata().unwrap();
 println!("Metadata:\n {}", Metadata::pretty_format(&meta.metadata).unwrap());
 ```